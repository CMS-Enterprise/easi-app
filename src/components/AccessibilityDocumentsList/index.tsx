import React, { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useTable } from 'react-table';
import { Link, Table } from '@trussworks/react-uswds';

import {
  AccessibilityRequestDocumentCommonType,
  AccessibilityRequestDocumentStatus
} from 'types/graphql-global-types';
import { translateDocumentType } from 'utils/accessibilityRequest';
import formatDate from 'utils/formatDate';

type Document = {
  status: AccessibilityRequestDocumentStatus;
  url: string;
  uploadedAt: string;
  documentType: {
    commonType: AccessibilityRequestDocumentCommonType;
    otherTypeDescription: string | null;
  };
};

type DocumentsListProps = {
  documents: Document[];
  requestName: string;
};

const AccessibilityDocumentsList = ({
  documents,
  requestName
}: DocumentsListProps) => {
  const { t } = useTranslation('accessibility');

  const getDocType = (documentType: {
    commonType: AccessibilityRequestDocumentCommonType;
    otherTypeDescription: string;
  }) => {
    if (documentType.commonType !== 'OTHER') {
      return translateDocumentType(
        documentType.commonType as AccessibilityRequestDocumentCommonType
      );
    }
    return documentType.otherTypeDescription;
  };

  const columns: any = useMemo(() => {
    return [
      {
        Header: t('documentTable.header.documentName'),
        accessor: 'documentType',
        Cell: ({ value }: any) => getDocType(value)
      },
      {
        Header: t('documentTable.header.uploadedAt'),
        accessor: 'uploadedAt',
        Cell: ({ value }: any) => {
          if (value) {
            return formatDate(value);
          }
          return '';
        },
        width: '25%'
      },
      {
        Header: t('documentTable.header.actions'),
        Cell: ({ row }: any) => (
          <>
            {row.original.status === 'PENDING' && (
              <em>Virus scan in progress...</em>
            )}
            {row.original.status === 'AVAILABLE' && (
              <Link
                target="_blank"
                rel="noreferrer"
                href={row.original.url}
                aria-label={`View ${getDocType(
                  row.original.documentType
                )} in a new tab or window`}
              >
                {t('documentTable.view')}
              </Link>
            )}
            {row.original.status === 'UNAVAILABLE' && (
              <>
                <i className="fa fa-exclamation-circle text-secondary" />{' '}
                Document failed virus scan
              </>
            )}
            {/* <UswdsLink asCustom={Link} to="#" className="margin-left-2">
              {t('documentTable.remove')}
            </UswdsLink>
            <span className="usa-sr-only">{row.original.name}</span> */}
          </>
        )
      }
    ];
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow
  } = useTable({
    columns,
    data: documents,
    documents,
    initialState: {
      sortBy: [{ id: 'uploadedAt', desc: true }]
    }
  });

  if (documents.length === 0) {
    return <span>{t('requestDetails.documents.none')}</span>;
  }

  return (
    <Table bordered={false} {...getTableProps()} fullWidth>
      <caption className="usa-sr-only">
        {`${t('documentTable.caption')} ${requestName}`}
      </caption>
      <thead>
        {headerGroups.map(headerGroup => (
          <tr {...headerGroup.getHeaderGroupProps()}>
            {headerGroup.headers.map(column => (
              <th
                {...column.getHeaderProps()}
                style={{ whiteSpace: 'nowrap', width: column.width }}
                scope="col"
              >
                {column.render('Header')}
              </th>
            ))}
          </tr>
        ))}
      </thead>
      <tbody {...getTableBodyProps()}>
        {rows.map(row => {
          prepareRow(row);
          return (
            <tr {...row.getRowProps()}>
              {row.cells.map((cell, i) => {
                if (i === 0) {
                  return (
                    <th
                      {...cell.getCellProps()}
                      scope="row"
                      style={{ maxWidth: '16rem' }}
                    >
                      {cell.render('Cell')}
                    </th>
                  );
                }
                return (
                  <td {...cell.getCellProps()} style={{ maxWidth: '16rem' }}>
                    {cell.render('Cell')}
                  </td>
                );
              })}
            </tr>
          );
        })}
      </tbody>
    </Table>
  );
};

export default AccessibilityDocumentsList;
