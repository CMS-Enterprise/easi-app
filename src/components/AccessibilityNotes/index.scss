.accessibility-note {
  &__note-item {
    display: flex;
    &:not(:last-of-type) {
      border-bottom: 1px solid #979797;
      margin-bottom: 1rem;
    }
  }
}