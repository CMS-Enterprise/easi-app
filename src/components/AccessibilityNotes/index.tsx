import React from 'react';
import { DateTime } from 'luxon';

import './index.scss';

type AccessibilityNote = {
  content: string;
  author: string;
  createdAt: string;
  id: string;
};

export const AccessibilityNoteListItem = ({
  note
}: {
  note: AccessibilityNote;
}) => {
  return (
    <div className="accessibility-note__note-item" id={note.id}>
      <div className="margin-bottom-2">
        <p className="margin-top-0 margin-bottom-1 text-pre-wrap">
          {note.content}
        </p>
        <span className="text-base-dark font-body-2xs">{`by ${
          note.author
        } | ${DateTime.fromISO(note.createdAt).toLocaleString(
          DateTime.DATE_FULL
        )}`}</span>
      </div>
    </div>
  );
};

export const AccessibilityNotesList = ({
  notes
}: {
  notes: AccessibilityNote[];
}) => {
  return (
    <div>
      <caption className="usa-sr-only">{`${notes.length} existing notes`}</caption>
      {notes.map(note => (
        <AccessibilityNoteListItem note={note} />
      ))}
    </div>
  );
};
