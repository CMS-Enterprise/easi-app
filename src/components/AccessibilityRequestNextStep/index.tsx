import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Link } from '@trussworks/react-uswds';

// import './index.scss';

const AccessibilityRequestNextStep = () => {
  const { t } = useTranslation('accessibility');

  return (
    <div className="margin-bottom-3">
      <h2 className="margin-y-0 font-heading-lg">
        {t('requestDetails.documents.noDocs.heading')}
      </h2>
      <p className="line-height-body-4">
        <Trans i18nKey="accessibility:requestDetails.documents.noDocs.description">
          indexZero
          <Link
            className="display-inline-block"
            target="_blank"
            rel="noopener noreferrer"
            href="/508/templates"
          >
            linkText
          </Link>
          indexOne
        </Trans>
      </p>
    </div>
  );
};

export default AccessibilityRequestNextStep;
