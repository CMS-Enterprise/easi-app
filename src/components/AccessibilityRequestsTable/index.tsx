/* eslint-disable react/prop-types */

import React, { FunctionComponent, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { useSortBy, useTable } from 'react-table';
import { Link as UswdsLink, Table } from '@trussworks/react-uswds';
import classnames from 'classnames';
import { DateTime } from 'luxon';

import formatDate from 'utils/formatDate';

// import cmsDivisionsAndOffices from 'constants/enums/cmsDivisionsAndOffices';

type AccessibilityRequestsTableRow = {
  id: string;
  system: {
    name: string;
    businessOwner: {
      name?: string;
      component?: string;
    };
  };
  submittedAt?: DateTime;
  testDate?: DateTime;
  relevantTestDate?: {
    date: DateTime;
  };
  status?: string;
  lastUpdatedAt?: DateTime;
};

type AccessibilityRequestsTableProps = {
  requests: AccessibilityRequestsTableRow[];
};

const AccessibilityRequestsTable: FunctionComponent<AccessibilityRequestsTableProps> = ({
  requests
}) => {
  const { t } = useTranslation('accessibility');
  const columns: any = useMemo(() => {
    return [
      {
        Header: t('requestTable.header.requestName'),
        accessor: 'name',
        Cell: ({ row, value }: any) => {
          return (
            <UswdsLink asCustom={Link} to={`/508/requests/${row.original.id}`}>
              {value}
            </UswdsLink>
          );
        },
        maxWidth: 350
      },
      {
        Header: t('requestTable.header.submissionDate'),
        accessor: 'submittedAt',
        Cell: ({ value }: any) => {
          if (value) {
            return formatDate(value);
          }
          return '';
        }
      },
      {
        Header: t('requestTable.header.businessOwner'),
        accessor: (row: AccessibilityRequestsTableRow) => {
          return `${row.system.businessOwner.name}, ${row.system.businessOwner.component}`;
        }
      },
      {
        Header: t('requestTable.header.testDate'),
        accessor: (row: AccessibilityRequestsTableRow) => {
          return `${row.relevantTestDate?.date || 'Not Added'}`;
        }
      }
      // {
      //   Header: t('requestTable.header.status'),
      //   accessor: 'status',
      //   Cell: ({ row, value }: any) => {
      //     const date = DateTime.fromISO(row.original.updatedAt).toLocaleString(
      //       DateTime.DATE_FULL
      //     );
      //     return (
      //       <>
      //         <strong>{value}</strong>
      //         <br />
      //         <span>{`${t('requestTable.lastUpdated')} ${date}`}</span>
      //       </>
      //     );
      //   }
      // }
    ];
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow
  } = useTable(
    {
      columns,
      data: requests,
      sortTypes: {
        alphanumeric: (rowOne, rowTwo, columnName) => {
          const rowOneElem = rowOne.values[columnName];
          const rowTwoElem = rowTwo.values[columnName];

          // If item is a string, enforce capitalization (temporarily) and then compare
          if (typeof rowOneElem === 'string') {
            return rowOneElem.toUpperCase() > rowTwoElem.toUpperCase() ? 1 : -1;
          }

          // If item is a DateTime, convert to Number and compare
          if (rowOneElem instanceof DateTime) {
            return Number(rowOneElem) > Number(rowTwoElem) ? 1 : -1;
          }

          // If neither string nor DateTime, return bare comparison
          return rowOneElem > rowTwoElem ? 1 : -1;
        }
      },
      requests,
      initialState: {
        sortBy: [{ id: 'submittedAt', desc: true }]
      }
    },
    useSortBy
  );

  const getHeaderSortIcon = (isDesc: boolean | undefined) => {
    return classnames('margin-left-1', {
      'fa fa-caret-down': isDesc,
      'fa fa-caret-up': !isDesc
    });
  };

  const getColumnSortStatus = (
    column: any
  ): 'descending' | 'ascending' | 'none' => {
    if (column.isSorted) {
      if (column.isSortedDesc) {
        return 'descending';
      }
      return 'ascending';
    }

    return 'none';
  };

  return (
    <Table bordered={false} {...getTableProps()} fullWidth>
      <caption className="usa-sr-only">{t('requestTable.caption')}</caption>
      <thead>
        {headerGroups.map(headerGroup => (
          <tr {...headerGroup.getHeaderGroupProps()}>
            {headerGroup.headers.map(column => (
              <th
                {...column.getHeaderProps(column.getSortByToggleProps())}
                aria-sort={getColumnSortStatus(column)}
                style={{ whiteSpace: 'nowrap' }}
              >
                {column.render('Header')}
                {column.isSorted && (
                  <span className={getHeaderSortIcon(column.isSortedDesc)} />
                )}
              </th>
            ))}
          </tr>
        ))}
      </thead>
      <tbody {...getTableBodyProps()}>
        {rows.map(row => {
          prepareRow(row);
          return (
            <tr {...row.getRowProps()}>
              {row.cells.map((cell, i) => {
                if (i === 0) {
                  return (
                    <th
                      {...cell.getCellProps()}
                      scope="row"
                      style={{ maxWidth: '16rem' }}
                    >
                      {cell.render('Cell')}
                    </th>
                  );
                }
                return (
                  <td {...cell.getCellProps()} style={{ maxWidth: '16rem' }}>
                    {cell.render('Cell')}
                  </td>
                );
              })}
            </tr>
          );
        })}
      </tbody>
    </Table>
  );
};

export default AccessibilityRequestsTable;
