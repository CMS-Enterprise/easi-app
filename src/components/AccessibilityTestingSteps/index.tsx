import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import {
  Link as UswdsLink,
  ProcessList,
  ProcessListHeading,
  ProcessListItem
} from '@trussworks/react-uswds';
import classNames from 'classnames';

import UswdsReactLink from 'components/LinkWrapper';
import CollapsableLink from 'components/shared/CollapsableLink';
import { ArticleComponentProps } from 'types/articles';

const AccessibilityTestingSteps = ({
  helpArticle,
  className
}: ArticleComponentProps) => {
  const { t } = useTranslation('accessibility');
  const exceptionReasons: string[] = t(
    'testingStepsOverview.exception.reasons',
    {
      returnObjects: true
    }
  );
  return (
    <div
      className={classNames(
        '508-steps-involved tablet:grid-col-6 margin-top-6',
        className
      )}
    >
      <ProcessList>
        <ProcessListItem>
          <ProcessListHeading type="h3">
            {t('testingStepsOverview.fillForm.heading')}
          </ProcessListHeading>
          <p>{t('testingStepsOverview.fillForm.description')}</p>
        </ProcessListItem>
        <ProcessListItem>
          <ProcessListHeading type="h3">
            {t('testingStepsOverview.prepareVPAT.heading')}
          </ProcessListHeading>
          <Trans
            i18nKey="accessibility:testingStepsOverview.prepareVPAT.fillOutVPAT"
            className="margin-0"
          >
            indexZero
            <UswdsReactLink
              to="/508/templates"
              target="_blank"
              rel="noopener noreferrer"
            >
              templatesLink
            </UswdsReactLink>
            indexTwo
          </Trans>
          <p className="margin-bottom-0">
            {t('testingStepsOverview.prepareVPAT.changesVPAT')}
          </p>
        </ProcessListItem>
        <ProcessListItem>
          <ProcessListHeading type="h3">
            {t('testingStepsOverview.testingSession.heading')}
          </ProcessListHeading>
          <p>{t('testingStepsOverview.testingSession.description')}</p>
        </ProcessListItem>
        <ProcessListItem>
          <ProcessListHeading type="h3">
            {t('testingStepsOverview.results.heading')}
          </ProcessListHeading>
          <p>{t('testingStepsOverview.results.description')}</p>
          <dl title="508 test scores">
            <dt className="text-bold margin-bottom-1">
              {t('testingStepsOverview.results.score.above99.heading')}
            </dt>
            <dd className="margin-left-0 margin-bottom-2">
              {t('testingStepsOverview.results.score.above99.description')}
            </dd>
            <dt className="text-bold margin-bottom-1">
              {t('testingStepsOverview.results.score.interval75.heading')}
            </dt>
            <dd className="margin-left-0 margin-bottom-2">
              {t('testingStepsOverview.results.score.interval75.description')}
            </dd>
            <dt className="text-bold margin-bottom-1">
              {t('testingStepsOverview.results.score.below75.heading')}
            </dt>
            <dd className="margin-left-0">
              {t('testingStepsOverview.results.score.below75.description')}
            </dd>
          </dl>
        </ProcessListItem>
      </ProcessList>
      <CollapsableLink
        id="easi-508-testing-exception"
        label={t('testingStepsOverview.exception.label')}
        styleLeftBar={false}
      >
        <div className="line-height-body-5">
          <p className="margin-top-0">
            {t('testingStepsOverview.exception.description')}
          </p>
          <ul>
            {exceptionReasons.map(reason => (
              <li key={reason}>{reason}</li>
            ))}
          </ul>
          <p>{t('testingStepsOverview.exception.exceptionFineprint')}</p>
          <p>
            <Trans i18nKey="accessibility:testingStepsOverview.exception.contact">
              indexZero
              <UswdsLink href="mailto:CMS_Section508@cms.hhs.gov">
                email
              </UswdsLink>
              indexTwo
            </Trans>
          </p>
        </div>
      </CollapsableLink>
    </div>
  );
};

export default AccessibilityTestingSteps;
