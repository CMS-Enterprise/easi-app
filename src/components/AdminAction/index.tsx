import React from 'react';
import { useTranslation } from 'react-i18next';
import { Button, ButtonGroup } from '@trussworks/react-uswds';
import classNames from 'classnames';

import UswdsReactLink from 'components/LinkWrapper';

type StateLink = {
  pathname: string;
  state: {
    fromAdmin: boolean;
  };
};

export type AdminActionButton = {
  label: string;
  link?: string | StateLink;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  outline?: boolean;
  unstyled?: boolean;
};

type AdminActionProps = {
  title: string;
  buttons: AdminActionButton[];
  description?: string;
  children?: React.ReactNode;
  className?: string;
};

/**
 * Admin action component for TRB admin home subpages
 */
const AdminAction = ({
  title,
  description,
  children,
  buttons,
  className
}: AdminActionProps) => {
  const { t } = useTranslation('technicalAssistance');

  return (
    <div className={classNames('usa-summary-box', className)}>
      <h5 className="text-base-dark text-uppercase margin-top-0 margin-bottom-05 line-height-body-1 text-normal text-body-xs">
        {t('adminAction.title')}
      </h5>
      <h3 className="margin-y-0">{t(title)}</h3>
      {description && (
        <p className="margin-y-0 line-height-body-5">{t(description)}</p>
      )}
      {children}
      <ButtonGroup className="margin-top-3">
        {buttons.map(button =>
          button.onClick ? (
            <Button
              key={button.label}
              type="button"
              onClick={button.onClick}
              outline={button.outline}
              unstyled={button.unstyled}
            >
              {button.label}
            </Button>
          ) : (
            <UswdsReactLink
              key={button.label}
              to={button.link || ''}
              className={classNames('usa-button', {
                'usa-button--unstyled': button.unstyled,
                'usa-button--outline': button.outline
              })}
            >
              {button.label}
            </UswdsReactLink>
          )
        )}
      </ButtonGroup>
    </div>
  );
};

export default AdminAction;
