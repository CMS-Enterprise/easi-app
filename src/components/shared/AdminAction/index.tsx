import React from 'react';
import { useTranslation } from 'react-i18next';
import { Button, ButtonGroup } from '@trussworks/react-uswds';
import classNames from 'classnames';

export type AdminActionButton = {
  label: string;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  outline?: boolean;
  unstyled?: boolean;
};

type AdminActionProps = {
  title: string;
  buttons: AdminActionButton[];
  description?: string;
  children?: React.ReactNode;
  className?: string;
};

/**
 * Admin action component for TRB admin home subpages
 */
const AdminAction = ({
  title,
  description,
  children,
  buttons,
  className
}: AdminActionProps) => {
  const { t } = useTranslation('technicalAssistance');

  return (
    <div className={classNames('usa-summary-box', className)}>
      <h5 className="text-base-dark text-uppercase margin-top-0 margin-bottom-05 line-height-body-1 text-normal text-body-xs">
        {t('adminAction.title')}
      </h5>
      <h3 className="margin-y-0">{t(title)}</h3>
      {description && (
        <p className="margin-y-0 line-height-body-5">{t(description)}</p>
      )}
      {children}
      <ButtonGroup className="margin-top-3">
        {buttons.map(button => (
          <Button
            key={button.label}
            type="button"
            onClick={button.onClick}
            outline={button.outline}
            unstyled={button.unstyled}
          >
            {button.label}
          </Button>
        ))}
      </ButtonGroup>
    </div>
  );
};

export default AdminAction;
