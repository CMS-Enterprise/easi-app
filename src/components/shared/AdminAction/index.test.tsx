import React from 'react';
import { render } from '@testing-library/react';
import i18next from 'i18next';

import { ActionText } from 'views/TechnicalAssistance/AdminHome/components/TrbAdminWrapper';

import AdminAction from '.';

describe('admin action component', () => {
  const text: ActionText = i18next.t(
    'technicalAssistance:adminAction.statuses.REQUEST_FORM_COMPLETE',
    { returnObjects: true }
  );
  it('Matches the snapshot', () => {
    const { asFragment } = render(
      <AdminAction
        title={text.title}
        description={text.description}
        buttons={[
          {
            label: i18next.t(
              'technicalAssistance:adminAction.buttons.requestEdits'
            ),
            onClick: () => null,
            outline: true
          },
          {
            label: i18next.t(
              'technicalAssistance:adminAction.buttons.readyForConsult'
            ),
            onClick: () => null,
            outline: true
          },
          {
            label: i18next.t(
              'technicalAssistance:adminAction.buttons.orCloseRequest'
            ),
            onClick: () => null,
            unstyled: true
          }
        ]}
      >
        {text.list?.label}
        <ul>
          {(text.list?.unorderedItems || []).map(item => (
            <li key={item}>{item}</li>
          ))}
        </ul>
      </AdminAction>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
