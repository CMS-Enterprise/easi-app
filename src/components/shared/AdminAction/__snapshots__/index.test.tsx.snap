// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin action component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="usa-summary-box"
  >
    <h5
      class="text-base-dark text-uppercase margin-top-0 margin-bottom-05 line-height-body-1 text-normal text-body-xs"
    >
      Admin Action
    </h5>
    <h3
      class="margin-y-0"
    >
      Is this request ready for a consult?
    </h3>
    <p
      class="margin-y-0 line-height-body-5"
    >
      Review the initial request form. If this request is ready to move on to a consult session, please mark it as ready. If the TRB needs further information in order to proceed, you may request edits or additional information from the project team.
    </p>
    What should I consider when reviewing the initial request form?
    <ul>
      <li>
        Is the support requested clear, i.e. design review vs. problem-solving?
      </li>
      <li>
        If the customer is asking for assistance in solving a problem, is the problem definition clear?
      </li>
      <li>
        Is the stated scope of the changes to the project clear?
      </li>
      <li>
        If the scope is clear, are the architecture changes minor (i..e. simple or straightforward)?
      </li>
      <li>
        Is there enough information available to determine which SMEs are needed?
      </li>
      <li>
        Has the project been to a TRB consult session before?
      </li>
      <li>
        If the project has been to a TRB consult session before, are the stakeholders, and the contractor the same as those that attended before?
      </li>
    </ul>
    <ul
      class="usa-button-group margin-top-3"
    >
      <li
        class="usa-button-group__item"
      >
        <button
          class="usa-button usa-button--outline"
          data-testid="button"
          type="button"
        >
          Request edits
        </button>
      </li>
      <li
        class="usa-button-group__item"
      >
        <button
          class="usa-button usa-button--outline"
          data-testid="button"
          type="button"
        >
          Ready for consult
        </button>
      </li>
      <li
        class="usa-button-group__item"
      >
        <button
          class="usa-button usa-button--unstyled"
          data-testid="button"
          type="button"
        >
          or, close this request
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
